augly/video/functional.py

```diff
@@ -138,6 +141,64 @@ def audio_swap(
     return output_path or video_path
 
 
+def augment_audio(
+    video_path: str,
+    audio_aug_function: Callable[..., Tuple[np.ndarray, int]] = audaugs.apply_lambda,
+    output_path: Optional[str] = None,
+    metadata: Optional[List[Dict[str, Any]]] = None,
+    **audio_aug_kwargs,
+) -> str:
+    """
+    Augments the audio track of the input video using a given AugLy audio augmentation
+
+    @param video_path: the path to the video to be augmented
+
+    @param aug_function: the augmentation function to be applied onto the video's audio
+        track. Should have the standard API of an AugLy audio augmentation, i.e. expect
+        input audio as a numpy array or path & output path as input, and output the 
+        augmented audio to the output path
+
+    @param output_path: the path in which the resulting video will be stored.
+        If not passed in, the original video file will be overwritten
+
+    @param metadata: if set to be a list, metadata about the function execution
+        including its name, the source & dest duration, fps, etc. will be appended
+        to the inputted list. If set to None, no metadata will be appended or returned
+
+    @param audio_aug_kwargs: the input attributes to be passed into `audio_aug`
+
+    @returns: the path to the augmented video
+    """
+    assert callable(audio_aug_function), (
+        repr(type(audio_aug_function).__name__) + " object is not callable"
+    )
+
+    func_kwargs = helpers.get_func_kwargs(
+        metadata, locals(), video_path, audio_aug_function=audio_aug_function.__name__
+    )
+
+    audio_metadata = []
+    with tempfile.NamedTemporaryFile(suffix=".wav") as tmpfile:
+        helpers.extract_audio_to_file(video_path, tmpfile.name)
+        print("name ", audio_aug_function.__name__)
+        print("kwargs ",**audio_aug_kwargs)
+        aug_audio, aug_sr = audio_aug_function(
+            tmpfile.name, metadata=audio_metadata, **audio_aug_kwargs
+        )
```

Comment:
I forgot to add this change that you need to make in order for `augment_audio` to work with a transform class augmentation as well as a functional augmentation. Since we changed the unit test to use a transform class, that's why it was failing.
```suggestion
        audio, sr = audutils.validate_and_load_audio(tmpfile.name)
        aug_audio, aug_sr = audio_aug_function(
            audio, sample_rate=sr, metadata=audio_metadata, **audio_aug_kwargs
```